src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -161,6 +166,1134 @@ void CodeGen::genSSE2Intrinsic(GenTreeHWIntrinsic* node)
+                case TYP_DOUBLE:
+                    ins = INS_andnpd;
+                    break;
+                default:
+                    unreached();
+                    break;
+            }
+
+            emit->emitIns_SIMD_R_R_R(ins, targetReg, op1Reg, op2Reg, TYP_SIMD16);
+            break;
+        }
+
+        case NI_SSE2_Average:
+        {
+            op2Reg = op2->gtRegNum;
+
+            instruction ins;
+            switch (baseType)
+            {
+                case TYP_UBYTE:
+                    ins = INS_pavgb;
+                    break;
+                case TYP_USHORT:
+                    ins = INS_pavgw;
+                    break;
+                default:
+
+                    unreached();
+                    break;
+            }
+
+            emit->emitIns_SIMD_R_R_R(ins, targetReg, op1Reg, op2Reg, TYP_SIMD16);
+            break;
+        }
+
+        case NI_SSE2_CompareEqual:
+        {
+            op2Reg = op2->gtRegNum;
+
+            instruction ins;
+            switch (baseType)
+            {
+                case TYP_DOUBLE:
+                    // Instruction with immediate byte operand controlling comparison type
+                    ins = INS_cmppd;
+                    emit->emitIns_SIMD_R_R_R_I(ins, targetReg, op1Reg, op2Reg, 0, TYP_SIMD16);
+                    break;
+                case TYP_INT:
+                case TYP_UINT:
+                    ins = INS_pcmpeqd;
+                    break;
+                case TYP_BYTE:
+                case TYP_UBYTE:
+                    ins = INS_pcmpeqb;
+                    break;
+                case TYP_SHORT:
+                case TYP_USHORT:
+                    ins = INS_pcmpeqw;
+                    break;
+                default:
+                    unreached();
+                    break;
+            }
+
+            if (ins != INS_cmppd)
+            {
+                emit->emitIns_SIMD_R_R_R(ins, targetReg, op1Reg, op2Reg, TYP_SIMD16);
+            }
+            break;
+        }
+
+        case NI_SSE2_CompareGreaterThan:
+        case NI_SSE2_CompareNotLessThanOrEqual:
+        {
+            op2Reg = op2->gtRegNum;
+
+            instruction ins;
+            switch (baseType)
+            {
+                case TYP_DOUBLE:
+                    // Instruction with immediate byte operand controlling comparison type
+                    ins = INS_cmppd;
+                    emit->emitIns_SIMD_R_R_R_I(ins, targetReg, op1Reg, op2Reg, 6, TYP_SIMD16);
+                    break;
+                case TYP_INT:
+                    ins = INS_pcmpgtd;
+                    break;
+                case TYP_BYTE:
+                    ins = INS_pcmpgtb;
+                    break;
+                case TYP_SHORT:
+                    ins = INS_pcmpgtw;
+                    break;
+                default:
+                    unreached();
+                    break;
+            }
+
+            if (ins != INS_cmppd)
```

Comment:
you could drop this check if you return instead of break in the TYP_DOUBLE case.